src/main/java/com/endava/webapp/controller/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,62 @@
+package com.endava.webapp.controller;
+
+import com.endava.webapp.exception.ErrorResponse;
+import com.endava.webapp.exception.ResourceNotFoundException;
+import com.endava.webapp.exception.UniqueConstraintViolationException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+
+@RestControllerAdvice
+public class ControllerExceptionHandler {
+
+    @ResponseBody
+    @ExceptionHandler(ResourceNotFoundException.class)
+    public ResponseEntity<ErrorResponse> resourceNotFoundHandler(
+            ResourceNotFoundException e, WebRequest request) {
+        ErrorResponse errorResponse = new ErrorResponse(e.getMessage(), request.getDescription(false),
+                HttpStatus.NOT_FOUND.value(), HttpStatus.NOT_FOUND.getReasonPhrase());
+        return new ResponseEntity<>(errorResponse, HttpStatus.NOT_FOUND);
+    }
+
+    @ResponseBody
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<ErrorResponse> internalError(Exception e, WebRequest request) {
+        ErrorResponse errorResponse = new ErrorResponse(e.getMessage(), request.getDescription(false),
+                HttpStatus.INTERNAL_SERVER_ERROR.value(), HttpStatus.INTERNAL_SERVER_ERROR.getReasonPhrase());
+        return new ResponseEntity<>(errorResponse, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @ResponseBody
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<ErrorResponse> methodArgumentNotValidHandler(
+            MethodArgumentNotValidException e, WebRequest request) {
+        ErrorResponse errorResponse = new ErrorResponse(e.getMessage(), request.getDescription(false),
+                HttpStatus.BAD_REQUEST.value(), HttpStatus.BAD_REQUEST.getReasonPhrase());
+        return new ResponseEntity<>(errorResponse, HttpStatus.BAD_REQUEST);
+    }
+
+    @ResponseBody
+    @ExceptionHandler(MethodArgumentTypeMismatchException.class)
+    public ResponseEntity<ErrorResponse> methodArgumentNotValidHandler(
+            MethodArgumentTypeMismatchException e, WebRequest request) {
+        ErrorResponse errorResponse = new ErrorResponse(e.getMessage(), request.getDescription(false),
+                HttpStatus.BAD_REQUEST.value(), HttpStatus.BAD_REQUEST.getReasonPhrase());
+        return new ResponseEntity<>(errorResponse, HttpStatus.BAD_REQUEST);
+    }
+
+    // TODO - finish exception handling
```

Comment:
As I see you uncommented other exc. handlers, so probably we could remove this comment